core/commands/diff.py

```diff
@@ -213,12 +213,29 @@ def run(self, edit, setting):
         self.view.window().status_message("{} is now {}".format(setting, settings.get(setting_str)))
 
         self.view.run_command("gs_diff_refresh")
-        if last_cursors:
-            sel = self.view.sel()
-            sel.clear()
-            for (a, b) in last_cursors:
-                sel.add(sublime.Region(a, b))
-            self.view.show(sel)
+        if setting == 'in_cached_mode':
+            if self.view.settings().get("git_savvy.diff_view.just_hunked"):
```

Comment:
Just from a stylistic viewpoint, seems like this behaviour along with `last_cursors` getter/setter on lines 207-209 belong together. For readability, I would recommend moving these together into their own method (e.g `move_to_hunk`).

Reply:
We have the `if setting == 'in_cached_mode':` three times in this function so it is better to extract this functionality https://github.com/divmain/GitSavvy/pull/1075/commits/a6c67faa56bbb57acefb1a70957594523f32e428 to actually reduce complexity.